### PR TITLE
fix some frequent warnings

### PR DESCRIPTION
**Pull request by cbielow**
## Description

...

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Review by timosachsenberg — APPROVED** [review]

**Review by jpfeuffer — APPROVED** [review]
much better thanks

File: src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h [hunk 1]
```diff
@@ -175,7 +176,7 @@ namespace OpenMS
         }
 
         out_it->setPosition(first->getPosition());
-        out_it->setIntensity(std::max(0.0, help));
+        out_it->setIntensity(std::max<IntensityType>(0, help));
```
Comment (jpfeuffer) [inline src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h:179]: Wait. Why the different usage here?

Comment (jpfeuffer) [reply src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h:179]: compared to before

File: src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h [hunk 2]
```diff
@@ -121,7 +121,8 @@ namespace OpenMS
       int i;
       UInt j;
       int mid = (frame_size_ / 2);
-      double help;
+      using IntensityType = typename InputIt::value_type::IntensityType;
+      IntensityType help;
```
Comment (cbielow) [inline src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h:125]:
```suggestion
      double help;
```

File: src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h [hunk 3]
```diff
@@ -137,7 +138,7 @@ namespace OpenMS
         }
 
         out_it->setPosition(first->getPosition());
-        out_it->setIntensity(std::max(0.0, help));
+        out_it->setIntensity(std::max(IntensityType{}, help));
```
Comment (cbielow) [inline src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h:141]:
```suggestion
        out_it->setIntensity(std::max(0.0, help));
```

File: src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h [hunk 4]
```diff
@@ -157,7 +158,7 @@ namespace OpenMS
...
```
Comment (cbielow) [inline src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h:161]:
```suggestion
        out_it->setIntensity(std::max(0.0, help));
```

File: src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h (hunk 1 above)
Comment (cbielow) [inline src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h:179]:
```suggestion
        out_it->setIntensity(std::max(0.0, help));
```

**Review by cbielow — COMMENTED** [review]
reverting changes to SG since it seems to break on MacOSX

File: src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h (hunk 1 above)
Comment (cbielow) [reply src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h]: no particular reason. But needs reverting anyway. No idea why it breaks... we have IEEE math .. but not on MacOS :)

**Comment by cbielow** [comment]
MacOS passes.
Will merge.

**Pull request merged**